### PR TITLE
Add byte array output binding

### LINKED ISSUE #25810: azure functions event grid out binding not working with typescript [BUG]
### Library name and version

Microsoft.Azure.WebJobs.Extensions.EventGrid 3.0.0

### Describe the bug

I have a simple typescript azure function with an eventGrid out binding. When the function finishes I want an array of events to be sent using event grid. This works fine with the extension bundle version 2.x, but when I upgrade to version 3.x I run into issues. Version 3.x of the extensions bundle has an updated version of Microsoft.Azure.WebJobs.Extensions.EventGrid compared to version 2.x.

### Expected behavior

I expect the function to finish and send my events using event grid. This works fine with the extension bundle version 2.x.

### Actual behavior

The error I get is this:
`System.Private.CoreLib: Exception while executing function: Microsoft.Azure.WebJobs.Host: Error while handling parameter _binder after function returned:. Microsoft.Azure.WebJobs.Extensions.EventGrid: System.Byte[] is not a valid event type.`

### Reproduction Steps

Make a simple azure function using typescript, node 14 and azure functions runtime version 4.  Use the extension bundle version 3.x (latest). Add an event grid out binding and try to add some event objects to the context to send them when the function finishes.

### Environment

Testing locally:
Core Tools Version: 4.0.3971.
Function Runtime Version: 4.0.1.16815.
Node: Version 14 og 16, I get the error using both.
ExtensionBundle version: [3.3.0, 4.0.0)

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Fixes https://github.com/Azure/azure-sdk-for-net/issues/25810

### CODE REVIEW

**Review by jsquire — APPROVED** [review]

File: sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/OutputBinding/EventGridAsyncCollector.cs [hunk 1]
```diff
@@ -51,128 +51,83 @@ public async Task FlushAsync(CancellationToken cancellationToken = default(Cance
...
-                    {
-                        var ev = EventGridEvent.Parse(new BinaryData(str));
-                        isEventGridEvent = true;
-                    }
-                    catch (ArgumentException)
-                    {
-                    }
-
-                    if (isEventGridEvent)
+                    case string:
```
Comment (pakrym) [inline sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/OutputBinding/EventGridAsyncCollector.cs:56]:
```suggestion
                    case string s:
```

Comment (pakrym) [reply sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/OutputBinding/EventGridAsyncCollector.cs:56]: to avoid the extra cast

Comment (pakrym) [reply sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/OutputBinding/EventGridAsyncCollector.cs:56]: You can also pull await SendAsync out of the switch

```
BinaryData bd = evt switch {
...
};
```

Comment (JoshLove-msft) [reply sdk/eventgrid/Microsoft.Azure.WebJobs.Extensions.EventGrid/src/OutputBinding/EventGridAsyncCollector.cs:56]: Actually, that is just the first event, so we still need to do the cast for the other events.

**Pull request merged**